File: src/main/java/com/pedigree/util/DirtyFlag.java
package com.pedigree.util;

public final class DirtyFlag {
    private static volatile boolean modified = false;

    private DirtyFlag() { }

    public static boolean isModified() {
        return modified;
    }

    public static void setModified() {
        modified = true;
    }

    public static void clear() {
        modified = false;
    }
}

File: src/main/kotlin/com/family/tree/util/DirtyFlag.kt
package com.family.tree.util

/**
 * Global dirty flag indicating that in-memory project state has been modified
 * and needs persistence. Thread visibility is ensured via @Volatile.
 */
object DirtyFlag {
    @Volatile
    private var modified: Boolean = false

    @JvmStatic
    fun isModified(): Boolean = modified

    @JvmStatic
    fun setModified() {
        modified = true
    }

    @JvmStatic
    fun clear() {
        modified = false
    }
}
